### PR TITLE
Erto 118 lazy loading carousels

### PR DESCRIPTION
**Pull request by xendk**
Ensure carousels are loading content lazily, and fix styling to avoid load jumping.

### CODE REVIEW

File: sites/all/modules/paragraphs/reol_news_paragraph/reol_news_paragraph.module [hunk 1]
```diff
@@ -22,29 +22,143 @@ function reol_news_paragraph_field_formatter_info() {
 }
 
 /**
+ * Implements hook_menu().
+ */
+function reol_news_paragraph_menu() {
+  $items = array();
+
+  $items['reol_news_paragraph/ajax/%/%/%/%/%'] = array(
```
Comment (danquah) [inline sites/all/modules/paragraphs/reol_news_paragraph/reol_news_paragraph.module]: A comment or two explaining what these arguments are about would be nice

File: sites/all/modules/paragraphs/reol_news_paragraph/reol_news_paragraph.module [hunk 2]
```diff
@@ -22,29 +22,143 @@ function reol_news_paragraph_field_formatter_info() {
...
+ * @param string $field_name
+ *   Name of the field.
+ * @param string $delta
+ *   Field delta.
+ * @param int $offset
+ *   Offset into result to fetch from.
+ */
+function reol_news_paragraph_ajax($entity_type, $entity_id, $field_name, $delta, $offset) {
+  // The access callback has already checked that these exists.
+  $entity = entity_load($entity_type, array($entity_id));
```
Comment (danquah) [inline sites/all/modules/paragraphs/reol_news_paragraph/reol_news_paragraph.module]: Does it make sense to handle the situation where the entity does not load?

Comment (danquah) [reply sites/all/modules/paragraphs/reol_news_paragraph/reol_news_paragraph.module]: (no idea whether the check has already been made one way or the other before the callback is hit)

File: sites/all/modules/paragraphs/reol_news_paragraph/reol_news_paragraph.module [hunk 3]
```diff
@@ -22,29 +22,143 @@ function reol_news_paragraph_field_formatter_info() {
...
+    $item = $items[$delta];
+    $articles = _reol_news_paragraph_get_articles($item['value']);
+    foreach ($articles as $article) {
+      $build[] = array(
+        '#type' => 'ding_carousel_item',
+        '#content' => $article,
+      );
+    }
+
+    $return['content'] = drupal_render($build);
```
Comment (danquah) [inline sites/all/modules/paragraphs/reol_news_paragraph/reol_news_paragraph.module]: build is potentially null/unset here.

File: sites/all/modules/paragraphs/reol_news_paragraph/reol_news_paragraph.module (hunk 2 above)
Comment (xendk) [reply sites/all/modules/paragraphs/reol_news_paragraph/reol_news_paragraph.module]: As *the comment above* says, the access callback for the page has already done the exact same thing.

File: sites/all/modules/paragraphs/reol_news_paragraph/reol_news_paragraph.module (hunk 1 above)
Comment (xendk) [reply sites/all/modules/paragraphs/reol_news_paragraph/reol_news_paragraph.module]: They're documented on the function, which grabs them all? I don't think it's common to document arguments in hook_menu().

File: sites/all/modules/paragraphs/reol_news_paragraph/reol_news_paragraph.module (hunk 3 above)
Comment (xendk) [reply sites/all/modules/paragraphs/reol_news_paragraph/reol_news_paragraph.module]: Fixed.

**Pull request merged**